lib/entry/FlavorDownloader.js

```diff
@@ -119,41 +120,43 @@ var FlavorDownloader = (function() {
         var that = this;
         return Q.allSettled(_.map(chunksToDownload, function(chunkName) {
             var chunkUrl = url.resolve(baseUrl + '/', chunkName);
-
-            //return the promise of the downloaded chunk
-            return that.tsHttpUtils.downloadConvert(chunkUrl, path.join(that.destFolderPath, chunkName))
-                .then(function(mp4File) {
-                    mp4File.flavor = that.flavor;
-                    mp4File.chunkName = chunkName;
-                    delete that.chunksToDownload[chunkName];
-                    that.downloadedChunks[chunkName] = true;
-                    //create link between original ts chunk name and mp4 file name
-                    that.downloadedChunks[path.basename(mp4File.path)] = chunkName;
-                    return mp4File;
-                })
-                .catch(function(err){
-                    // do not postpone download of the next chunks in line when chunk is corrupted
-                    if(err instanceof MP4ConversionError){
-                        that.logger.warn("failed to convert: error string: %s, chunk: %s", err,err.chunkName);
-                        delete that.chunksToDownload[chunkName];
-                        return Q.resolve(null);
-                    }
+            return fsUtils.createHierarchyPath(that.destFolderPath)
```

Comment:
Was thinking about it when wrote it. Still testing to see performance results. In general, makeTree only completes the paths that do not exist, so there's no recreation of the directory with each call.